lib/handler/chunked.c

```diff
@@ -49,15 +49,15 @@ static void send_chunk(h2o_ostream_t *_self, h2o_req_t *req, h2o_iovec_t *inbufs
         memcpy(outbufs + outbufcnt, inbufs, sizeof(h2o_iovec_t) * inbufcnt);
         outbufcnt += inbufcnt;
         outbufs[outbufcnt].base = "\r\n0\r\n\r\n";
-        outbufs[outbufcnt].len = is_final ? 7 : 2;
+        outbufs[outbufcnt].len = state == H2O_SEND_STATE_FINAL ? 7 : 2;
```

Comment:
I think you would need to set `req->http1_is_persistent` to zero to avoid the connection from being used in case of sending an incomplete chunked response.

Also, we might want to emit "\r\n1\r\n" (a non-zero chunk size) but without following data in this case, taking https://github.com/h2o/h2o/pull/1031#issuecomment-241485544 into consideration.

The same applies to the below `if` as well.